Content.Server/Blob/BlobCoreSystem.cs

```diff
@@ -0,0 +1,338 @@
+            return;
+
+        if (newChem == component.CurrentChem)
+            return;
+
+        var oldChem = component.CurrentChem;
+        component.CurrentChem = newChem;
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+
+            blobTileComponent.Color = component.ChemСolors[newChem];
+            Dirty(blobTileComponent);
+
+            if (TryComp<BlobFactoryComponent>(blobTile, out var blobFactoryComponent))
+            {
+                if (TryComp<BlobbernautComponent>(blobFactoryComponent.Blobbernaut, out var blobbernautComponent))
+                {
+                    blobbernautComponent.Color = component.ChemСolors[newChem];
+                    Dirty(blobbernautComponent);
+
+                    if (TryComp<MeleeWeaponComponent>(blobFactoryComponent.Blobbernaut, out var meleeWeaponComponent))
+                    {
+                        var blobbernautDamage = new DamageSpecifier();
+                        foreach (var keyValuePair in component.ChemDamageDict[component.CurrentChem].DamageDict)
+                        {
+                            blobbernautDamage.DamageDict.Add(keyValuePair.Key, keyValuePair.Value * 0.8f);
+                        }
+                        meleeWeaponComponent.Damage = blobbernautDamage;
+                    }
+
+                    ChangeBlobEntChem(blobFactoryComponent.Blobbernaut.Value, oldChem, newChem);
+                }
+
+                foreach (var compBlobPod in blobFactoryComponent.BlobPods)
+                {
+                    if (TryComp<SmokeOnTriggerComponent>(compBlobPod, out var smokeOnTriggerComponent))
+                    {
+                        smokeOnTriggerComponent.SmokeColor = component.ChemСolors[newChem];
+                    }
+                }
+            }
+
+            ChangeBlobEntChem(blobTile, oldChem, newChem);
+        }
+    }
+
+    private void OnDestruction(EntityUid uid, BlobCoreComponent component, DestructionEventArgs args)
+    {
+        if (component.Observer != null)
+        {
+            QueueDel(component.Observer.Value);
+        }
+
+        foreach (var blobTile in component.BlobTiles)
+        {
+            if (!TryComp<BlobTileComponent>(blobTile, out var blobTileComponent))
+                continue;
+            blobTileComponent.Core = null;
+
+            blobTileComponent.Color = Color.White;
+            Dirty(blobTileComponent);
+        }
+    }
+
+    private void ChangeBlobEntChem(EntityUid uid, BlobChemType oldChem, BlobChemType newChem)
+    {
+        var explosionResistance = EnsureComp<ExplosionResistanceComponent>(uid);
+        if (oldChem == BlobChemType.ExplosiveLattice)
+        {
+            _explosionSystem.SetExplosionResistance(uid, 0.3f, explosionResistance);
+        }
+        switch (newChem)
+        {
+            case BlobChemType.ExplosiveLattice:
```

Comment:
Why are the blob types enums instead of being a prototype.

Please make a prototype for defining blob types instead of hardcoding them.